catalog/nsili/catalog-nsili-common/src/main/java/org/codice/alliance/nsili/common/ResultDAGConverter.java

```diff
@@ -84,6 +84,53 @@
   private static final Pattern ATTRIBUTE_PATTERN =
       Pattern.compile("([a-zA-Z0-9_:]+):([a-zA-Z0-9_]+).([a-zA-Z0-9]+)");
 
+  private static Map<String, String> typeConversionMap;
+
+  private static boolean forceHttp;
+
+  static {
+    typeConversionMap = new HashMap<>();
+    typeConversionMap.put("Collection", "COLLECTION/EXPLOITATION PLAN");
+    // Text can be any of DOCUMENT, MESSAGE, CBRN, and others - will be checked later
+    typeConversionMap.put("Text", "DOCUMENT");
+    // typeConversionMap.put("Text", "GEOGRAPHIC AREA OF INTEREST");
+    // typeConversionMap.put("", "GEOSPATIAL VECTOR");
+    // typeConversionMap.put( "", "GMTI");
+    typeConversionMap.put("Image", "IMAGERY");
+    // typeConversionMap.put("", "INTELLIGENCE REQUIREMENT");
+    // typeConversionMap.put("Text", "MESSAGE");
+    // typeConversionMap.put("Text", "OPERATIONAL ROLES"});
+    // typeConversionMap.put("Text", "ORBAT"});
+    // typeConversionMap.put("Text", "REPORT"});
+    // typeConversionMap.put("", "RFI"});
+    // typeConversionMap.put("Text", "SYSTEM ASSIGNMENTS");
+    // typeConversionMap.put("Text", "SYSTEM SPECIFICATIONS");
+    // typeConversionMap.put("Text", "SYSTEM DEPLOYMENT STATUS");
+    // typeConversionMap.put("Text", "TACTICAL SYMBOL");
+    // typeConversionMap.put("", "TDL DATA");
```

Comment:
There are a lot of commented out items here. Should they be removed or associated with a future ticket?

Reply:
Probably converted to a future ticket - the mappings between metacard types and NSIL types is not one to one, so we need a smarter converter than just a table lookup. For example, we would need to identify the difference between various XML products and set the NSIL type accordingly.

Reply:
Created https://codice.atlassian.net/browse/CAL-520 to address this